controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -301,10 +306,141 @@ func (r *ReplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	err = r.processLastActionResult(ctx, localRG, remoteClient, log)
+	if err != nil {
+		r.EventRecorder.Eventf(localRG, eventTypeWarning, eventReasonUpdated,
+			"failed to process the last action %s", localRG.Status.LastAction.Condition)
+	}
+
 	log.V(common.InfoLevel).Info("RG has already been synced to the remote cluster")
 	return ctrl.Result{}, nil
 }
 
+func (r *ReplicationGroupReconciler) processLastActionResult(ctx context.Context, group *repv1.DellCSIReplicationGroup, remoteClient connection.RemoteClusterClient, log logr.Logger) error {
+	if group.Status.LastAction.ErrorMessage != "" {
```

Comment:
It's better to first check for any action presence (2nd if stmt), then check for error message on last action (1st if stmt). Please revert the order.

Reply:
Updated.